ToDoApp/main.py

```diff
@@ -9,5 +16,56 @@ async def root():
     return {"message": "Hello World"}
 
 
+@app.post("/user/register/", tags=["users"], response_model=str)
+async def register(*, user: schemas.UserCreate = Body(default=None), db: Session = Depends(get_DB),
+                   background_tasks: BackgroundTasks):
+    """
+    This function adds the user in DB and sends a verification email
+    :param user:
+    :param db:
+    :param background_tasks:
+    :return:
+    """
+    db_user = crud.get_user_by_email(db, user_email=user.email)
+    if db_user:
+        raise HTTPException(status_code=400, detail="Email already registered")
+    hashed_password = utils.hash_password(user.password)
+    user_db = schemas.UserCreate(email=user.email, name=user.name, password=hashed_password)
+    stored_user = crud.create_user(db=db, user=user_db)
+    token = utils.encode_token(stored_user.email)
+    # check if token already exists
+    db_token = crud.get_token(db, user_id=stored_user.id)
+    if not db_token:
+        crud.create_token(db, token=token, user_id=stored_user.id)
+        encrypted_token = utils.encrypt_token(token.access_token)
+        token = encrypted_token.decode("utf-8")
+        verification_link = f"http://localhost:8000/user/confirmation/{token}"
+        background_tasks.add_task(auth_email.send_verification_email, token=verification_link, user_email=stored_user.email)
+        return verification_link
+
+
+@app.get("/user/confirmation/{token}", tags=["users"])
+def confirmation(token: str, db: Session = Depends(get_DB)):
+    """
+    this function confirms the user
+    :param token:
+    :param db:
+    :return:
+    """
+    token = token.encode("utf-8")
+    decrypt_token = utils.decrypt_token(token)
+    decoded_token = utils.decode_token(decrypt_token)
+    if decoded_token:
+        user = crud.get_user_by_email(db, decoded_token.get("email"))
+        if user:
+            res = crud.confirmed_user(db, user.email)
+            if not res:
+                crud.confirmation(db, user.email)
+                return {"Message": "User Confirmed"}
+            raise HTTPException(status_code=400, detail="User already registered")
+        raise HTTPException(status_code=400, detail="Email already registered")
+    raise HTTPException(status_code=400, detail="Token expired")
```

Comment:
do you think this is more clean and easy to follow instead of nested IFs
```
    if not decoded_token:
        raise HTTPException(status_code=400, detail="Token expired")

    user = crud.get_user_by_email(db, decoded_token.get("email"))
    if not user:
        raise HTTPException(status_code=400, detail="Email already registered")

    res = crud.confirmed_user(db, user.email)
    if res:
        raise HTTPException(status_code=400, detail="User already registered")
    
    crud.confirmation(db, user.email)
    return {"Message": "User Confirmed"}
```

Reply:
Yes, this is cleaner. I have updated using this approach.